include/ts/apidefs.h.in

```diff
@@ -1472,6 +1472,18 @@ namespace ts
 }
 #endif
 
+///
+/// @brief JSON-RPC Handler options
+///
+/// This class holds information about how a handler will be managed and delivered when called. The JSON-RPC manager would use this
+/// object to perform certain validation.
+///
+typedef struct TSRPCHandlerOptions_s {
+  struct Auth {
+    int restricted; ///< Tells the RPC Manager if the call can be delivered or not based on the config rules.
+  } auth;
+} TSRPCHandlerOptions;
+
```

Comment:
Why is this in apidefs.h ?

Reply:
So can be directly used from both sides(core & plugins). isn't that the point of `apidefs.h`?

Reply:
> So you're saying it's for future use?

short answer: yes.

long asnwer:
The future is here:  https://github.com/apache/trafficserver/pull/8630,  I will update 8630  after this one is done.
At yahoo we are running the rpc TS API in production since quite some time now, so all this changes will get into our yahoo ats as well  very soon.